snakemake_executor_plugin_kubernetes/__init__.py

```
@@ -212,6 +217,38 @@ def run_job(self, job: JobExecutorInterface):
+        # ---------------------------------------------------------------------
+        if "node_taint" in resources_dict:
+            if pod_spec.tolerations is None:
+                pod_spec.tolerations = []
+            toleration = str(resources_dict["node_taint"])
+            key, rest = toleration.split("=", 1)
+            value, effect = rest.split(":", 1)
+            if key == "" or value == "" or effect == "":
+                raise WorkflowError(
+                    "Invalid node_taint format. "
+                    "Expected format: key=value:effect"
+                )
```

Ruff (0.14.10)

Avoid specifying long messages outside the exception class

(TRY003)